stl/inc/ranges

```diff
@@ -732,6 +736,46 @@ namespace ranges {
         bool _Engaged = false;
     };
 
+    template <_Destructible_object _Ty>
+        requires is_trivially_destructible_v<_Ty>
+    class _Non_propagating_cache<_Ty, false> { // a specialization for trivially destructible types where checking if
+                                               // the cache contains value is not needed
+    public:
+        constexpr _Non_propagating_cache() noexcept {}

```

Comment:
Can we default here?

Reply:
I don't know, but the original implementation doesn't have it `default`ed.

Reply:
Per [class.default.ctor]/2:

> A defaulted default constructor for class `X` is defined as deleted if:
(2.1) &mdash; `X` is a union that has a variant member with a non-trivial default constructor and no variant member
of `X` has a default member initializer,
> [...]

`= default` here would actually delete the default constructor if `_Ty` has a non-trivial default constructor.